src/screens/SendTokens.js

```diff
@@ -203,33 +215,73 @@ class SendTokens extends React.Component {
     this.setState({ errorMessage: message, ledgerStep: 0 });
   }
 
+  /**
+   * Method executed before sending a tx to be signed by Ledger
+   *
+   * Checks if we have any custom token, if not we just send to sign
+   * If we have custom tokens we must send all tokens first
+   * When all tokens are sent we should send the tx to sign
+   */
+  beforeSendLedger = () => {
+    // remove HTR if present
+    const txTokens = this.state.txTokens.filter(t => !hathorLib.tokens.isHathorToken(t.uid));
+
+    if (txTokens.length === 0) {
+      // no custom tokens, just send
+      this.executeSendLedger();
+      return;
+    }
+    // get all custom tokens, and include the signatures
+    const tokenSignatures = tokens.getTokenSignatures();
+    const missingSigs = txTokens.filter(t => !tokenSignatures.hasOwnProperty(t.uid));
+    if (missingSigs.length !== 0) {
+      // there are tokens without signatures, missingSigs
+      // set tittle and content
+      this.ledgerAlertModalTitle = t`Unverified custom tokens`
+      this.ledgerAlertModalBody = this.renderAlertTokenList(missingSigs);
+      $("#ledgerAlertModal").modal('show');
+      return;
+    }
+
+    const tokenSigs = txTokens.map(t => {
+      return {
+        uid: t.uid,
+        symbol: t.symbol,
+        name: t.name,
+        signature: tokenSignatures[t.uid],
+      };
+    });
+
+    ledger.sendTokens(tokenSigs);
+  }
+
   /**
    * Method executed to send transaction on ledger
    * It opens the ledger modal to wait for user action on the device
-   *
-   * @param {Object} data Transaction data
    */
-  executeSendLedger = (data) => {
+  executeSendLedger = () => {
     // Complete data with default values
-    hathorLib.transaction.completeTx(data);
+    hathorLib.transaction.completeTx(this.data);
+
+    this.sendTransaction = hathorLib.SendTransaction({ outputs: this.data.outputs, inputs: this.data.inputs, network: this.props.wallet.getNetworkObject() });
+    this.data = this.sendTransaction.prepareTxData();
 
     // Find if tx has change output
-    const changeIndex = data.outputs.findIndex((v) => v.isChange === true);
+    const changeIndex = this.data.outputs.findIndex((v) => v.isChange === true);
```

Comment:
With multiple tokens we can have more than one change output, one for each token. How are handling this in ledger? As far as I can see, `ledger.sendTx(this.data, changeIndex, changeKeyIndex);` method allows for only one change output, is that correct? Do we need to add this support on ledger as well?

Reply:
Yes, this is a problem we have to fix.
There was no way to keep backwards compatibility when adding multiple change outputs.
We need a separate project to update the "sign tx" method on ledger.

Reply:
We need to discuss this, even create another design if needed. I don't think we should merge this PR like this because the implementation has bugs in this part.